backend/recipe_endpoints.py

```diff
@@ -129,3 +129,36 @@ async def create_recipe(
         logger.error(f"Error creating recipe: {str(e)}")
         db.rollback()
         raise HTTPException(status_code=500, detail="Failed to create recipe")
+@router.put("/api/recipes/{recipe_id}/upload-photo")
+async def upload_recipe_photo(
+    recipe_id: int,
+    image: UploadFile = File(...),
+    db: Session = Depends(get_db)
+):
+    try:
+        db_recipe = db.query(Recipe).filter(Recipe.recipe_id == recipe_id).first()
+        if not db_recipe:
+            raise HTTPException(status_code=404, detail="Recipe not found")
+
+        if not image.content_type.startswith('image/'):
+            raise HTTPException(status_code=400, detail="File must be an image")
+
+        image_url = await upload_image_to_s3(image, RECIPE_IMAGES_FOLDER)
```

Comment:
first delete the current image before uploading a new. if we just upload a new one, we leave the old image in the s3 bucket forever yet it will never be used. these unused images will slowly pile up and consume space, raising our s3 bill.

Reply:
This is not so clear, do you mean I should put that option of the user deleting the image first? 
If not, do I go to the s3 bucket and delete it from there? Need more clarity on this please.

Reply:
done